### ISSUE TITLE
Not Found: Cannot GET /debian/fqdn 

### ISSUE THREAD

**Issue by arthurlutz** [cozy/cozy-self-hosting#2]
My self-hosted instance of cozy is running inside of a docker, it´s a debian based docker iirc. 

When I open https://cozy.lutz.im/#apps/self-hosting/ I get : 

```
Not Found: Cannot GET /debian/fqdn 
```

**Comment by jeekajoo** [comment]
same here

**Comment by frankrousseau** [comment]
ping @nledez

**Comment by clochix** [comment]
Could you check the Node version you use? self-hosting requires Node 4.

**Comment by jeekajoo** [comment]
hum.

```
# nodejs -v
v0.10.29
# node -v
v0.10.29
# dpkg -l | grep node
ii  libjs-node-uuid                   1.4.0-1                      all          simple, fast generation of RFC4122 UUIDs - JavaScript library
ii  node-abbrev                       1.0.5-2                      all          Get unique abbreviations for a set of strings - Node.js module
ii  node-ansi                         0.3.0-2                      all          Advanced ANSI formatting tool for Node.js
ii  node-ansi-color-table             1.0.0-1                      all          Color and format tables for ansi output - Node.js module
ii  node-archy                        0.0.2-1                      all          Pretty-print nested hierarchies module for Node.js
ii  node-async                        0.8.0-1                      all          higher-order functions and common patterns for asynchronous Javascript
ii  node-block-stream                 0.0.7-1                      all          Stream of fixed-size blocks, with zero-padding when necessary
ii  node-combined-stream              0.0.5-1                      all          Append streams one after another - module for Node.js
ii  node-cookie-jar                   0.3.1-1                      all          Cookie handling for HTTP clients - module for Node.js
ii  node-delayed-stream               0.0.5-1                      all          Buffer stream events for later handling - module for Node.js
ii  node-forever-agent                0.5.1-1                      all          HTTP agent supporting keep-alive requests - module for Node.js
ii  node-form-data                    0.1.0-1                      all          Create multipart/form-data streams module for Node.js
ii  node-fstream                      0.1.24-1                     all          Advanced filesystem streaming tools for Node.js
ii  node-fstream-ignore               0.0.6-2                      all          Directory reader configurable by .ignore module for Node.js
ii  node-github-url-from-git          1.1.1-1                      all          Convert github git or gist url to an http url - Node.js module
ii  node-glob                         4.0.5-1                      all          glob functionality for Node.js
ii  node-graceful-fs                  3.0.2-1                      all          drop-in replacement improving the Node.js fs module
ii  node-gyp                          0.12.2+ds-1                  all          Native addon build tool for Node.js
ii  node-inherits                     2.0.1-1                      all          Exposes inherits function from Node.js environment
ii  node-ini                          1.1.0-1                      all          ini format parser and serializer for Node.js
ii  node-json-stringify-safe          5.0.0-1                      all          JSON.stringify with circular references module for Node.js
ii  node-lockfile                     0.4.1-1                      all          Asynchronous file lock module for Node.js
ii  node-lru-cache                    2.3.1-1                      all          least-recently-used cache object for Node.js
ii  node-mime                         1.2.11-1                     all          library for mime-type mapping for Node.js
ii  node-minimatch                    1.0.0-1                      all          Convert glob expressions into RegExp objects for Node.js
ii  node-mkdirp                       0.5.0-1                      all          Recursively create directories - Node.js module
ii  node-mute-stream                  0.0.4-1                      all          Pass-through stream that can be muted module for Node.js
ii  node-node-uuid                    1.4.0-1                      all          simple, fast generation of RFC4122 UUIDs - Node module
ii  node-nopt                         3.0.1-1                      all          Command-line option parser for Node.js
ii  node-normalize-package-data       0.2.2-1                      all          Normalizes package metadata - Node.js module
ii  node-npmlog                       0.0.4-1                      all          Logger with custom levels and colored output for Node.js
ii  node-once                         1.1.1-1                      all          Run a function only once with this module for Node.js
ii  node-osenv                        0.1.0-1                      all          Environment settings lookup module for Node.js
ii  node-qs                           2.2.4-1                      all          Parse, stringify query strings for Node.js
ii  node-read                         1.0.5-1                      all          Read user input from stdin module for Node.js
ii  node-read-package-json            1.2.4-1                      all          Read package.json for npm module for Node.js
ii  node-request                      2.26.1-1                     all          simplified HTTP request client module for Node.js
ii  node-retry                        0.6.0-1                      all          Retry strategies for failed operations module for Node.js
ii  node-rimraf                       2.2.8-1                      all          Deep deletion (like rm -rf) module for Node.js
ii  node-semver                       2.1.0-2                      all          Semantic Versioning for Node.js
ii  node-sha                          1.2.3-1                      all          Check and get file or stream hashes - module for Node.js
ii  node-sigmund                      1.0.0-1                      all          Quick and dirty signatures for Objects module for Node.js
ii  node-slide                        1.1.4-1                      all          Simple chain and asyncMap flow control module for Node.js
ii  node-tar                          0.1.18-1                     all          read and write portable tar archives module for Node.js
ii  node-tunnel-agent                 0.3.1-1                      all          HTTP proxy tunneling agent module for Node.js
ii  node-underscore                   1.7.0~dfsg-1                 all          JavaScript's functional programming helper library - NodeJS
ii  node-which                        1.0.5-2                      all          Cross-platform 'which' module for Node.js
ii  nodejs                            0.10.29~dfsg-2               amd64        evented I/O for V8 javascript
ii  nodejs-dev                        0.10.29~dfsg-2               amd64        evented I/O for V8 javascript (development files)
ii  nodejs-legacy                     0.10.29~dfsg-2               all          evented I/O for V8 javascript (legacy symlink)
```

**Comment by clochix** [comment]
Well, I guess you have to upgrade to Node 4.x to get the application work.

**Comment by jeekajoo** [comment]
Well, I've been using`cozy-debian` for a long time now.

``` bash
# apt update
(....)
All packages are up to date.

# cozy_management show_diag
[OK] CouchDB
[OK] Cozy Controller
[OK] Cozy Reverse Proxy (https)
[OK] Cozy Reverse Proxy (http)
```

Should I do something like `cozy_management migrate_2_node4` to fix it? ping @nledez
Thank you

**Comment by jeekajoo** [comment]
So.. No it won't work.
New `cozy-debian` versions are not published on https://debian.cozycloud.cc/ since [1.17](https://github.com/cozy/cozy-debian/blob/master/debian/changelog) 
And package `cozy-apt-node-list` is needed to provide apt config to get node4.

EDIT: unless I update my apt config from 
`deb [arch=amd64] https://debian.cozycloud.cc/debian jessie main` to `deb [arch=amd64] https://debian.cozycloud.cc/debian jessie cozy`
lulz

**Comment by jeekajoo** [comment]
upgraded to 1.20 and launched a migration to node4 that went wrong.
http://paste.jeekajoo.eu/?1b5dc3beed24a67b#mICJNpEt5heB+0NWOP9kBXeNy6Ya7mMYsjNr1GGJc/M=
well, shit.

EDIT:
I finally got it without losing any data.
I purged some packages and fixed cozy package `dpkg -P npm nodejs-legacy nodejs-dev && apt-get install -f`
I edited /etc/passwd and fixed some home dirs.

```
-cozy-data-system:x:1001:1001::/home/cozy-data-system:/bin/sh
+cozy-data-system:x:1001:1001::/usr/local/cozy/apps:/bin/sh
```

reinstalled data-system manually because it was screwed
`cd /usr/local/cozy/apps/data-system/ && sudo -u cozy-data-system npm install`
I could not do it via `cozy-monitor uninstall data-system && cozy-monitor install data-system` because controller was unable to start properly because of data-system being screwed. chicken and egg problem.
then controller could boot properly and I uninstalled reinstalled all apps using cozy-monitor because it was broken.
sorry to say that, but I think you better have to be a sysadmin to use cozy for a long time.
